addons/stock_account/models/product_template.py

```diff
@@ -0,0 +1,127 @@
+                Real Price: The cost price displayed is the price of the last outgoing product (will be use in case of inventory loss for example).""",
+        string="Costing Method", copy=True)
+    cost_method = fields.Char(compute='_compute_cost_method', store=True)
+    property_stock_account_input_id = fields.Many2one(
+        'account.account',
+        string='Stock Input Account', oldname='property_stock_account_input',
+        domain=[('deprecated', '=', False)],
+        company_dependent=True,
+        help="When doing real-time inventory valuation, counterpart journal items for all incoming stock moves will be posted in this account, unless "
+             "there is a specific valuation account set on the source location. When not set on the product, the one from the product category is used.")
+    property_stock_account_output_id = fields.Many2one(
+        'account.account',
+        string='Stock Output Account', oldname='property_stock_account_output',
+        domain=[('deprecated', '=', False)],
+        company_dependent=True,
+        help="When doing real-time inventory valuation, counterpart journal items for all outgoing stock moves will be posted in this account, unless "
+             "there is a specific valuation account set on the destination location. When not set on the product, the one from the product category is used.")
+
+    @api.depends('property_cost_method')
+    def _compute_cost_method(self):
+        for product in self:
+            product.cost_method = product.property_cost_method or product.categ_id.property_cost_method
+
+    @api.depends('property_valuation')
+    def _compute_valuation_type(self):
+        for product in self:
+            product.valuation = product.property_valuation or product.categ_id.property_valuation
+
+    @api.onchange('type')
+    def _onchange_type_valuation(self):
+        if self.type != 'product':
+            self.valuation = 'manual_periodic'
+
+    @api.model
+    def create(self, vals):
+        if vals.get('cost_method'):
+            vals['property_cost_method'] = vals['cost_method']
+        if vals.get('valuation'):
+            vals['property_valuation'] = vals['valuation']
+        return super(ProductTemplate, self).create(vals)
+
+    @api.multi
+    def _get_product_accounts(self):
+        """ Add the stock accounts related to product to the result of super()
+        @return: dictionary which contains information regarding stock accounts and super (income+expense accounts)
+        """
+        accounts = super(ProductTemplate, self)._get_product_accounts()
+        accounts.update({
+            'stock_input': self.property_stock_account_input_id or self.categ_id.property_stock_account_input_categ_id,
+            'stock_output': self.property_stock_account_output_id or self.categ_id.property_stock_account_output_categ_id,
+            'stock_valuation': self.categ_id.property_stock_valuation_account_id or False,
+        })
```

Comment:
You also write above code like  : 
```
accounts[ 'stock_input'] = self.property_stock_account_input_id or self.categ_id.property_stock_account_input_categ_id
accounts[ 'stock_output'] = self.property_stock_account_output_id or self.categ_id.property_stock_account_output_categ_id
accounts[ 'stock_valuation'] = self.categ_id.property_stock_valuation_account_id or False
```
I think its looks more readable code :turtle: